### PR TITLE
Add Rust API to optimize COCOPageMapper performance

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

 - Ticket no. 117181
 - Migrate COCOPageMapper algorithm from Python to Rust implementation

| Python impl. | Rust impl. |
| - | - |
| stream | coco-rust |

This improvement shows that
1) The time from starting the Python script to creating the COCO page map is decreased from ~200 secs (Python impl.) to ~10 secs (Rust impl.).
2) Memory cost for constructing the COCO page map is decreased from ~350 Mb (Python impl.) to ~260 Mb (Rust impl.).

### How to test
Existing COCO tests with `stream=True` cover these changes.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [x] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### New features
 - Add tabular data import/export
   (<https://github.com/openvinotoolkit/datumaro/pull/1089>)
+- Add Rust API to optimize COCOPageMapper performance
+  (<https://github.com/openvinotoolkit/datumaro/pull/1120>)
```
Comment (wonjuleee) [inline CHANGELOG.md]: could you move this to the enhancement?

Comment (vinnamkim) [reply CHANGELOG.md]: Moved

**Review by bonhunko — APPROVED** [review]
LGTM!

**Review by wonjuleee — APPROVED** [review]
LGTM

**Pull request merged**